### ISSUE TITLE
How to turn off cSpell.allowCompoundWords for Python or CPP?

### ISSUE THREAD

**Issue by zzzeek** [streetsidesoftware/vscode-spell-checker#683]
v1.10.2 is installed.

`"cSpell.allowCompoundWords": false` is present in my settings.json but words such as:

    arraylength
    genrating

etc. are not flagged.   if i change them to completely mis-spelled, such as:

    arrablength
    genxfrating

then they are flagged, as are other normally mis-spelled words, so the plugin is installed and working, just not allowCompoundWords.

I can put the directive in the file itself like this:

    # cSpell:disableCompoundWords

then it works, but that seems strange?   this can't be set user-wide?  does it have to be language specific ?

**Comment by Jason3S** [comment]
It is a bit confusing, but the configuration goes from general to specific. `cSpell.allowCompundWords` is a general setting and is by default `false`.

But some programming language configurations set it to true.

Please try:
```
"cSpell.languageSettings": [
    {
        "languageId": "python,cpp",
        "allowCompoundWords": false
    }
]

```

If means, when the programming language is `python` or `cpp` turn off compound words.

**Comment by zzzeek** [comment]
OK that works, and now I see, even though I had looked before, that the "languageId" syntax is indeed documented.    now let's see how we got here.

1. The very top setting on the settings there's a big checkbox that has no language setting at all and says, "C Spell: Allow Compound Words", along with the description, the box is not checked by default, and from here we can conclude that this setting does nothing at all, is that correct?    should this checkbox be removed?  otherwise what is it used for?


2. in the language specific checkbox, which I definitely found, it says "C Spell: Language Settings
Define settings on a per programming language basis.", and then "edit settings.json".    If there was a *little hint* there as to where I'd find the syntax to use in settings.json, that would be really helpful!

thanks for the help!

**Comment by Jason3S** [comment]
I agree that the documentation is not the best an could use some work.

Here is a link to where the dictionaries define `allowCompoundWords `: [Search · allowCompoundWords](https://github.com/streetsidesoftware/cspell-dicts/search?q=allowCompoundWords)

**Comment by Jason3S** [comment]
There is a point in the near future where `allowCompoundWords` will be turned off by default for all languages.

It has cause too many issues.

**Comment by Jason3S** [comment]
Closing stale issues.
